build.sbt

```diff
@@ -14,8 +14,26 @@ val AkkaPersistenceCassandraVersion = "0.17"
 val ScalaTestVersion = "2.2.4"
 val JacksonVersion = "2.7.2"
 val CassandraAllVersion = "3.0.2"
+val GuavaVersion = "19.0"
+val MavenVersion = "3.3.9"
+val NettyVersion = "4.0.36"
+
+// NOTE ON DEPENDENCIES
+
+// Since we support maven, we need to support mavens somewhat unintuitive dependency resolution mechanism. If two
+// versions of the same library are requested in maven, maven doesn't resolve the most recent version, it resolves
+// the version that is nearest to the root in the dependency graph.  So, when we depend on play, which 4 descendents
+// down the tree brings in netty-http-codec 4.0.36 which brings in netty-handler 4.0.36, and we also depend on
+// netty-reactive-streams, which brings in netty-handler 4.0.33, since, that's much higher in the tree than the other
+// path to it, we 4.0.33, and netty-http-codec 4.0.36 is incompatible with netty-handler 4.0.33, and so, well,
+// thankyou maven.
```

Comment:
Wow - didn't know this is how Maven resolved ambiguos artefact's version. The comment is extremely useful, but I definitely see this as a can of worm, and we should be extremely cautious when upgrading versions of any dependency in the future.